### PR TITLE
fix: clarify error message for EGITNOPERMISSION

### LINKED ISSUE #1291: Clarify EGITNOPERMISSION message to mention possible erroneous URL or server down
When running `npx semantic-release --branch my_branch --repository-url http://my.gitlab.domain/my_group/my_repo.git --no-ci` while the server is down, or when the repo URL is wrong, `@semantic-release/gitlab` always throws a `EGITNOPERMISSION` error, which is erronous and can be confusing.

Here is an example:
```
[..]

[12:00:28 PM] [semantic-release] › ✖  The command "git ls-remote --heads origin master_scenario_at_16_17_45_252584" failed with the error message fatal: unable to access 'http://gitlab.my.localhost/esciara/autorelease-test-repo-2019-08-30.git/': Failed to connect to gitlab.my.localhost port 80: Connection refused
.
[12:00:28 PM] [semantic-release] › ✖  EGITNOPERMISSION The push permission to the Git repository is required.
semantic-release cannot push the version tag to the branch master_scenario_at_16_17_45_252584 on remote Git repository with URL http://gitlab-ci-token:[secure]@gitlab.my.localhost/esciara/autorelease-test-repo-2019-08-304.git.

Please refer to the authentication configuration documentation to configure the Git credentials on your CI environment and make sure the repositoryUrl is configured with a valid Git URL.

SemanticReleaseError: The push permission to the Git repository is required.
    at module.exports (/path/node_modules/semantic-release/lib/get-error.js:6:10)
    at run (/path/node_modules/semantic-release/index.js:77:11)
    at processTicksAndRejections (internal/process/task_queues.js:85:5) {
  name: 'SemanticReleaseError',
  code: 'EGITNOPERMISSION',
  details: '**semantic-release** cannot push the version tag to the branch `master_scenario_at_16_17_45_252584` on remote Git repository with URL `http://gitlab-ci-token:[secure]@gitlab.my.localhost/esciara/autorelease-test-repo-2019-08-304.git`.\n' +
    '\n' +
    'Please refer to the [authentication configuration documentation](https://github.com/semantic-release/semantic-release/blob/master/docs/usage/ci-configuration.md#authentication) to configure the Git credentials on your CI environment and make sure the [repositoryUrl](https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#repositoryurl) is configured with a [valid Git URL](https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols).',
  semanticRelease: true
}
```

Is there any way to have the status checked before throwing this error? There are other instance when it is thrown, and the problem is not a permission problem. Not knowledgeable enough for a solution.

### PR DESCRIPTION
**Pull request by knidarkness**
This PR is for #1291 and it updates the error message for EGITNOPERMISSION to reflect possibly invalid url/server downtime. 

Unfortunately, there is no way to detect if user doesn't have permissions to access repo or the repo itself does not exist. So to make that more clear, PR adds info that EGITNOPERMISSION error might be caused because of non-existent or currently offline repository.

-------

Fix #1291

### CODE REVIEW

**Comment by pvdlg** [comment]
Can you change it to the follwoing?
```js
  EGITNOPERMISSION: ({options}) => ({
    message: 'Cannot push to the Git repository.',
    details: `**semantic-release** cannot push the version tag to the branch \`${
      options.branch
    }\` on remote Git repository with URL \`${options.repositoryUrl}\`.

This can be caused by either:
 - a misconfiguration of the [repositoryUrl option](${linkify('docs/usage/configuration.md#repositoryurl')})
 - the repository being unavailable
 - or missing push permission for the user asssociated wit the [Git credentials on your CI environment](${linkify(
   'docs/usage/ci-configuration.md#authentication'
 )})`,
  }),
```

**Comment by knidarkness** [comment]
Done as you have described

File: lib/definitions/errors.js [hunk 1]
```diff
@@ -26,19 +26,20 @@ Please make sure to add the \`repositoryUrl\` to the [semantic-release configura
       'docs/usage/configuration.md'
     )}).`,
   }),
-  EGITNOPERMISSION: ({options}) => ({
-    message: 'The push permission to the Git repository is required.',
+  EEGITNOPERMISSION: ({options}) => ({
```
Comment (pvdlg) [inline lib/definitions/errors.js]: Please do not change the message identifier.

File: lib/definitions/errors.js [hunk 2]
```diff
@@ -26,19 +26,20 @@ Please make sure to add the \`repositoryUrl\` to the [semantic-release configura
...
-    )}) is configured with a [valid Git URL](https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols).`,
+This can be caused by either:
+ - a misconfiguration of the [repositoryUrl option](${linkify('docs/usage/configuration.md#repositoryurl')})
+ - the repository being unavailable
+ - or missing push permission for the user asssociated wit the [Git credentials on your CI environment](${linkify(
+   'docs/usage/ci-configuration.md#authentication'
+ )})`,
   }),
-  EINVALIDTAGFORMAT: ({tagFormat}) => ({
+  INVALIDTAGFORMAT: ({tagFormat}) => ({
```
Comment (pvdlg) [inline lib/definitions/errors.js]: Same here, do not change the message identifier

File: lib/definitions/errors.js (hunk 1 above)
Comment (knidarkness) [reply lib/definitions/errors.js]: Oh, I am terribly sorry, had a mishap with vim editing.

**Comment by pvdlg** [comment]
I merged it via #1314 to make minor changes and squash your commits

**Pull request closed without merging**